vm/src/serde/deserialize_program.rs

```diff
@@ -461,38 +461,41 @@ pub fn parse_program_json(
     })
 }
 
-pub fn parse_program(program: Program) -> ProgramJson {
-    let references = program
-        .shared_program_data
-        .reference_manager
-        .iter()
-        .map(|r| Reference {
-            value_address: ValueAddress {
-                offset1: r.offset1.clone(),
-                offset2: r.offset2.clone(),
-                dereference: r.dereference,
-                value_type: r.cairo_type.clone().unwrap_or_default(),
-            },
-            ap_tracking_data: r.ap_tracking_data.clone().unwrap_or_default(),
-            pc: None,
-        })
-        .collect::<Vec<_>>();
-
-    ProgramJson {
-        prime: program.prime().to_string(),
-        builtins: program.builtins.clone(),
-        data: program.shared_program_data.data.clone(),
-        identifiers: program.shared_program_data.identifiers.clone(),
-        hints: program.shared_program_data.hints.clone(),
-        attributes: program.shared_program_data.error_message_attributes.clone(),
-        debug_info: program
+impl From<Program> for ProgramJson {
+    fn from(program: Program) -> Self {
```

Comment:
Let's take a reference here. Since we're already cloning the contents there's no need to drop the old instance.